### ISSUE TITLE
[Bug]: Failed to query with shard leader and query timeout

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#16628]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20220424-b82f6a1a
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.1.0.dev33
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: 

    queryNode:
      replicas: 5
  config:
    common:
      retentionDuration: 40
    queryNode:
      gracefulTime: 0
```
```


### Current Behavior

1. Deploy cluster with 5 querynodes
2. Run test case
```Python
   @pytest.mark.tags(CaseLabel.L3)
    def test_load_replica_change(self):
        """
        target: test load replica change
        method: 1.load with replica 1
                2.load with a new replica number
                3.release collection
                4.load with a new replica
        expected: The second time successfully loaded with a new replica number
        """
        # create, insert
        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix))
        df = cf.gen_default_dataframe_data()
        insert_res, _ = collection_w.insert(df)
        assert collection_w.num_entities == ct.default_nb

        collection_w.load(replica_number=1)
        collection_w.query(expr=f"{ct.default_int64_field_name} in [0]")
        collection_w.load(replica_number=2)
        collection_w.get_replicas()

        collection_w.release()
        collection_w.load(replica_number=2)
        collection_w.query(expr=f"{ct.default_int64_field_name} in [0]")
        # replica_info = collection_w.get_repl
```
3. Query rpc timeout

query segment info
```
[segmentID: 432748826231046145
collectionID: 432748825864044545
partitionID: 432748825864044546
mem_size: 787572
num_rows: 1503
nodeID: 11
state: Sealed
, segmentID: 432748826244153345
collectionID: 432748825864044545
partitionID: 432748825864044546
mem_size: 784428
num_rows: 1497
nodeID: 11
state: Sealed
, segmentID: 432748826231046145
collectionID: 432748825864044545
partitionID: 432748825864044546
mem_size: 787572
num_rows: 1503
nodeID: 12
state: Sealed
, segmentID: 432748826244153345
collectionID: 432748825864044545
partitionID: 432748825864044546
mem_size: 784428
num_rows: 1497
nodeID: 12
state: Sealed
]
```
The replicas info:
```
Replica groups:
- Group: <group_id:432748552538816532>, <group_nodes:(12, 8)>, <shards:[Shard: <channel_name:by-dev-rootcoord-dml_0_432748825864044545v0>, <shard_leader:8>, <shard_nodes:[12]>, Shard: <channel_name:by-dev-rootcoord-dml_1_432748825864044545v1>, <shard_leader:12>, <shard_nodes:[12]>]>
- Group: <group_id:432748552538816531>, <group_nodes:(11, 5, 2)>, <shards:[Shard: <channel_name:by-dev-rootcoord-dml_0_432748825864044545v0>, <shard_leader:2>, <shard_nodes:[11]>, Shard: <channel_name:by-dev-rootcoord-dml_1_432748825864044545v1>, <shard_leader:5>, <shard_nodes:[11]>]>
```

**Server logs:**
[mic_logs.tar.gz](https://github.com/milvus-io/milvus/files/8549849/mic_logs.tar.gz)

### Expected Behavior

Query successfully

**Comment by ThreadDao** [comment]
Running the case on image `master-20220426-60f7fef` and load collection `rpc timeout` exeption
**Case:**
```Python
    def test_load_replica_change(self):
        # create, insert
        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix))
        df = cf.gen_default_dataframe_data()
        insert_res, _ = collection_w.insert(df)
        assert collection_w.num_entities == ct.default_nb

        collection_w.load(replica_number=1)
        collection_w.query(expr=f"{ct.default_int64_field_name} in [0]")
        collection_w.load(replica_number=2)
        collection_w.get_replicas()

        collection_w.release()
        collection_w.load(replica_number=2)
        collection_w.query(expr=f"{ct.default_int64_field_name} in [0]")
        # replica_info = collection_w.get_replicas()
```
**Case running result:**
```
[2022-04-26 15:31:06 - DEBUG - ci_test]: (api_request)  : [Collection] args: ['collection_SjAp7H6B', {
  auto_id: False
  description:
  fields: [{
    name: int64
    description:
    type: 5
    is_primary: True
    auto_id: False
  }, {
    name: float
    description:
    type: 10
  }, {
    name: float_vector
    description:
    type: 101
    params: {'dim': 128}
 ......, kwargs: {'consistency_level': 'Strong'} (api_request.py:55)
[2022-04-26 15:31:06 - DEBUG - ci_test]: (api_response) : <Collection>:
-------------
<name>: collection_SjAp7H6B
<partitions>: [{"name": "_default", "collection_name": "collection_SjAp7H6B", "description": ""}]
<description>:
<schema>: {
  auto_id: False
  description:
  fields: [{
    name: int64
    description:
    type: 5
    is_primary: True
    a......  (api_request.py:27)
[2022-04-26 15:31:06 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float                                       float_vector
0         0     0.0  [0.15553233340025516, 0.09057178923608876, 0.1...
1         1     1.0  [0.05357376424842747, 0.08181711328514268, 0.1...
2         2     2.0  [0.13243863776130413, 0.040398913163702935, 0....
3         3    ......, kwargs: {'timeout': 20} (api_request.py:55)
[2022-04-26 15:31:07 - DEBUG - ci_test]: (api_response) : (insert count: 3000, delete count: 0, upsert count: 0, timestamp: 432796353651802113)  (api_request.py:27)
[2022-04-26 15:31:10 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 20], kwargs: {} (api_request.py:55)
[2022-04-26 15:31:30 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-04-26 15:31:10.040497', 'gRPC error': '2022-04-26 15:31:30.041710'}> (decorators.py:81)
[2022-04-26 15:31:51 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-04-26 15:31:31.043175', 'gRPC error': '2022-04-26 15:31:51.044725'}> (decorators.py:81)
[2022-04-26 15:32:12 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-04-26 15:31:52.046179', 'gRPC error': '2022-04-26 15:32:12.047827'}> (decorators.py:81)
[2022-04-26 15:32:33 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-04-26 15:32:13.049279', 'gRPC error': '2022-04-26 15:32:33.050679'}> (decorators.py:81)
[2022-04-26 15:32:54 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-04-26 15:32:34.052061', 'gRPC error': '2022-04-26 15:32:54.052743'}> (decorators.py:81)
[2022-04-26 15:33:15 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-04-26 15:32:55.054215', 'gRPC error': '2022-04-26 15:33:15.055709'}> (decorators.py:81)
[2022-04-26 15:33:36 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-04-26 15:33:16.057186', 'gRPC error': '2022-04-26 15:33:36.058702'}> (decorators.py:81)
[2022-04-26 15:33:57 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-04-26 15:33:37.060143', 'gRPC error': '2022-04-26 15:33:57.061723'}> (decorators.py:81)
[2022-04-26 15:34:18 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-04-26 15:33:58.062635', 'gRPC error': '2022-04-26 15:34:18.063634'}> (decorators.py:81)
[2022-04-26 15:34:39 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-04-26 15:34:19.065024', 'gRPC error': '2022-04-26 15:34:39.066268'}> (decorators.py:81)
[2022-04-26 15:34:39 - ERROR - ci_test]: Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 41, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 82, in handler
    raise e
  File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 695, in load_collection
    response = rf.result()
  File "/usr/local/lib/python3.6/site-packages/grpc/_channel.py", line 744, in result
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1650987279.065489443","description":"Error received from peer ipv4:10.98.0.9:19530","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Deadline Exceeded","grpc_status":4}"
>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/milvus/tests/python_client/utils/api_request.py", line 22, in inner_wrapper
    res = func(*args, **kwargs)
  File "/home/milvus/tests/python_client/utils/api_request.py", line 56, in api_request
    return func(*arg, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pymilvus/orm/collection.py", line 435, in load
    conn.load_collection(self._name, timeout=timeout, replica_number=replica_number, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pymilvus/decorators.py", line 52, in handler
    raise MilvusException(Status.UNEXPECTED_ERROR, "rpc timeout")
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=rpc timeout)>
 (api_request.py:35)
[2022-04-26 15:34:39 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=rpc timeout)> (api_request.py:36)
```

**Server logs:** 
[mic_logs.tar.gz](https://github.com/milvus-io/milvus/files/8564743/mic_logs.tar.gz)

**Comment by ThreadDao** [comment]
Maybe fixed `master-20220428-68b1b82f`
